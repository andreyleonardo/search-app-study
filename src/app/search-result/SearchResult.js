import React, { Component } from 'react';
import PropTypes from 'prop-types';
import OutsideClickHandler from 'react-outside-click-handler';
import ResultCardTour from './result-card-tour/ResultCardTour';
import ResultDetailsTour from './result-details-tour/ResultDetailsTour';
import pictures from '../../pictures';

const descriptionStub = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';

class SearchResult extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showDetails: false
    };

    this.showDetails = this.showDetails.bind(this);
    this.onOutsideClick = this.onOutsideClick.bind(this);
  }

  onOutsideClick() {
    const { showDetails } = this.state;
    if (showDetails) {
      this.setState({ showDetails: false });
    }
  }

  showDetails() {
    this.setState(state => ({ showDetails: !state.showDetails }));
  }

  render() {
    const {
      result: {
        title,
        rating,
        currency,
        price,
        isSpecialOffer,
      },
      tabIndex,
      index
    } = this.props;
    const { showDetails } = this.state;

    return (
      <OutsideClickHandler onOutsideClick={this.onOutsideClick}>
        <div id={`search-result-id-${index}`} className="search_result row center-xs">
          <section className="col-xs-12 col-sm-7">
            <ResultCardTour
              index={index}
              title={title}
              rating={parseFloat(rating)}
              currency={currency}
              price={price}
              isSpecialOffer={isSpecialOffer}
              onClick={this.showDetails}
              tabIndex={tabIndex}
              imgSrc={pictures[index].img}
              imgAlt={pictures[index].credits}
            />
          </section>
          <section className="col-xs-12 col-sm-7">
            <ResultDetailsTour
              index={index}
              title={title}
              description={descriptionStub}
              expand={showDetails}
              imgSrc={pictures[index].img}
              imgAlt={pictures[index].credits}
            />
          </section>
        </div>
      </OutsideClickHandler>
    );
  }
}


SearchResult.propTypes = {
  result: PropTypes.shape({
    title: PropTypes.string.isRequired,
    rating: PropTypes.string,
    currency: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    isSpecialOffer: PropTypes.bool
  }).isRequired,
  tabIndex: PropTypes.number,
  index: PropTypes.number
};

SearchResult.defaultProps = {
  tabIndex: 0,
  index: 0
};

export default SearchResult;
