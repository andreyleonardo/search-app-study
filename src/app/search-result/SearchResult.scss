.search_result {
  &__info {
    background-color: white;
    flex-basis: 100%;
    max-height: 0px;
    overflow: hidden;
    margin: 3px;
    transition: all .3s ease-in-out;
  }

  &__info--expanded {
    padding: 0.2em;
    max-height: 550px;
  }
}
