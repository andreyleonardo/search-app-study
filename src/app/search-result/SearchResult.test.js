import React from 'react';
import SearchResult from './SearchResult';

describe('Search Result', () => {
  let searchResult;
  beforeEach(() => {
    const props = {
      result: {
        title: 'My Title',
        currency: '$',
        price: '30',
        rating: '0'
      }
    };
    searchResult = shallow(<SearchResult {...props} />);
  });
  it('renders without crashing', () => {
    expect(searchResult).toMatchSnapshot();
  });
});
