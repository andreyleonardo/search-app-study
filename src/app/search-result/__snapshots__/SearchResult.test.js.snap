// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Result renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResult
    index={0}
    result={
      Object {
        "currency": "$",
        "price": "30",
        "rating": "0",
        "title": "My Title",
      }
    }
    tabIndex={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="search_result row center-xs"
        id="search-result-id-0"
      >
        <section
          className="col-xs-12 col-sm-7"
        >
          <ResultCardTour
            currency="$"
            imgAlt="Photo by Vladimir Kudinov on Unsplash"
            imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
            index={0}
            isSpecialOffer={false}
            onClick={[Function]}
            price="30"
            rating={0}
            tabIndex={0}
            title="My Title"
          />
        </section>
        <section
          className="col-xs-12 col-sm-7"
        >
          <ResultDetailsTour
            description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
            expand={false}
            imgAlt="Photo by Vladimir Kudinov on Unsplash"
            imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
            index={0}
            title="My Title"
          />
        </section>
      </div>,
      "disabled": false,
      "display": "block",
      "onOutsideClick": [Function],
      "useCapture": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="col-xs-12 col-sm-7"
          >
            <ResultCardTour
              currency="$"
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              isSpecialOffer={false}
              onClick={[Function]}
              price="30"
              rating={0}
              tabIndex={0}
              title="My Title"
            />
          </section>,
          <section
            className="col-xs-12 col-sm-7"
          >
            <ResultDetailsTour
              description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
              expand={false}
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              title="My Title"
            />
          </section>,
        ],
        "className": "search_result row center-xs",
        "id": "search-result-id-0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ResultCardTour
              currency="$"
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              isSpecialOffer={false}
              onClick={[Function]}
              price="30"
              rating={0}
              tabIndex={0}
              title="My Title"
            />,
            "className": "col-xs-12 col-sm-7",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "currency": "$",
              "imgAlt": "Photo by Vladimir Kudinov on Unsplash",
              "imgSrc": "vladimir-kudinov-58771-unsplash.jpeg",
              "index": 0,
              "isSpecialOffer": false,
              "onClick": [Function],
              "price": "30",
              "rating": 0,
              "tabIndex": 0,
              "title": "My Title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ResultDetailsTour
              description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
              expand={false}
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              title="My Title"
            />,
            "className": "col-xs-12 col-sm-7",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
              "expand": false,
              "imgAlt": "Photo by Vladimir Kudinov on Unsplash",
              "imgSrc": "vladimir-kudinov-58771-unsplash.jpeg",
              "index": 0,
              "title": "My Title",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="search_result row center-xs"
          id="search-result-id-0"
        >
          <section
            className="col-xs-12 col-sm-7"
          >
            <ResultCardTour
              currency="$"
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              isSpecialOffer={false}
              onClick={[Function]}
              price="30"
              rating={0}
              tabIndex={0}
              title="My Title"
            />
          </section>
          <section
            className="col-xs-12 col-sm-7"
          >
            <ResultDetailsTour
              description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
              expand={false}
              imgAlt="Photo by Vladimir Kudinov on Unsplash"
              imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
              index={0}
              title="My Title"
            />
          </section>
        </div>,
        "disabled": false,
        "display": "block",
        "onOutsideClick": [Function],
        "useCapture": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="col-xs-12 col-sm-7"
            >
              <ResultCardTour
                currency="$"
                imgAlt="Photo by Vladimir Kudinov on Unsplash"
                imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
                index={0}
                isSpecialOffer={false}
                onClick={[Function]}
                price="30"
                rating={0}
                tabIndex={0}
                title="My Title"
              />
            </section>,
            <section
              className="col-xs-12 col-sm-7"
            >
              <ResultDetailsTour
                description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
                expand={false}
                imgAlt="Photo by Vladimir Kudinov on Unsplash"
                imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
                index={0}
                title="My Title"
              />
            </section>,
          ],
          "className": "search_result row center-xs",
          "id": "search-result-id-0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ResultCardTour
                currency="$"
                imgAlt="Photo by Vladimir Kudinov on Unsplash"
                imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
                index={0}
                isSpecialOffer={false}
                onClick={[Function]}
                price="30"
                rating={0}
                tabIndex={0}
                title="My Title"
              />,
              "className": "col-xs-12 col-sm-7",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "currency": "$",
                "imgAlt": "Photo by Vladimir Kudinov on Unsplash",
                "imgSrc": "vladimir-kudinov-58771-unsplash.jpeg",
                "index": 0,
                "isSpecialOffer": false,
                "onClick": [Function],
                "price": "30",
                "rating": 0,
                "tabIndex": 0,
                "title": "My Title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ResultDetailsTour
                description="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
                expand={false}
                imgAlt="Photo by Vladimir Kudinov on Unsplash"
                imgSrc="vladimir-kudinov-58771-unsplash.jpeg"
                index={0}
                title="My Title"
              />,
              "className": "col-xs-12 col-sm-7",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
                "expand": false,
                "imgAlt": "Photo by Vladimir Kudinov on Unsplash",
                "imgSrc": "vladimir-kudinov-58771-unsplash.jpeg",
                "index": 0,
                "title": "My Title",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "section",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
